src/main/scala/strawman/collection/mutable/ArrayDeque.scala

```diff
@@ -0,0 +1,292 @@
+package scala
+package collection.mutable
```

Comment:
@DarkDimius : Another alternative would be to include this as a private collection in scala 2.12.3 and change the underlying implementation of `mutable.{Queue, Stack, Buffer}` to simply proxy to this - since this is strictly better than all of those (I have some hacky benchmarks but not a full suite to post yet).